doc/hooks/waf.section.md

```diff
@@ -1,3 +1,36 @@
 # wafHook {#wafhook}
 
-Overrides the configure, build, and install phases. This will run the “waf” script used by many projects. If `wafPath` (default `./waf`) doesn’t exist, it will copy the version of waf available in Nixpkgs. `wafFlags` can be used to pass flags to the waf script.
+[Waf](https://waf.io) is a Python-based software building system.
+
+In Nixpkgs, `wafHook` overrides the default configure, build, and install phases.
+
+## Variables controlling wafHook
+
+### `wafPath` {#wafPath}
+
+Location of the `waf` tool. It defaults to `./waf`, to honor software projects that include it directly inside their source trees.
+
+If `wafPath` doesn't exist, then `wafHook` will copy the `waf` provided from Nixpkgs to it.
+
+### `wafConfigureFlags` {#wafConfigureFlags}
+
+Controls the flags passed to waf tool during configure phase.
+
+### `wafFlags` {#wafFlags}
+
+Controls the flags passed to waf tool during build and install phases.
+
+### `dontAddWafCrossFlags` {#dontAddWafCrossFlags}
+
+When set to `true`, don't add cross compilation flags during configure phase.
+
+### Variables honored by wafHook {#variablesHonoredByWafHook}
+
+- `prefixKey`
+- `configureTargets`
+- `enableParallelBuilding`
+- `enableParallelInstalling`
+- `buildFlags`
+- `buildTargets`
+- `installFlags`
+- `installTargets`
```

Comment:
Waf uses `make` under the hood? Perhaps that's worth mentioning as well.

Reply:
I read now the rest of `waf/setup-hook.sh`, and I don't see a reason why would we support for instance both `installFlags` and `wafFlags`. How about supporting only `wafInstallFlags` and `wafBuildFlags`?

Reply:
You say, all flags should be  `waf`-prefixed?  Why?

The `wafFlags` are common  flags to be passed to build and install.

On the other hand, the script is a bit inconsistent.